From 137d8a19f06c0f59d36030c4fc97bda43085ceca Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Wed, 23 Nov 2016 16:49:10 +0200
Subject: [PATCH 1123/2241] fix: clk: ap806: fix typo in clock name, dclk
 instead of hclk

Fixed all affected drivers and files.
- axim driver
- Documentation
- system-controller driver

Change-Id: I0a51705ddede29e89ffdace9e1a7b9d4e79474bc
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34002
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 .../arm/marvell/ap806-system-controller.txt        |  4 ++--
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      |  6 +++---
 drivers/clk/mvebu/ap806-system-controller.c        | 24 +++++++++++-----------
 3 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/Documentation/devicetree/bindings/arm/marvell/ap806-system-controller.txt b/Documentation/devicetree/bindings/arm/marvell/ap806-system-controller.txt
index e951241..938e1c0 100644
--- a/Documentation/devicetree/bindings/arm/marvell/ap806-system-controller.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/ap806-system-controller.txt
@@ -24,7 +24,7 @@ Required properties:
  - reg: register area of the AP806 system controller
  - #clock-cells: must be set to 1
  - clock-output-names: must be defined to:
-   "ap-cpu-cluster-0", "ap-cpu-cluster-1", "ap-fixed", "ap-mss", "ap-emmc", "ap-hclk"
+   "ap-cpu-cluster-0", "ap-cpu-cluster-1", "ap-fixed", "ap-mss", "ap-emmc", "ap-dclk"
 
 Example:
 
@@ -32,6 +32,6 @@ Example:
 		compatible = "marvell,ap806-system-controller", "syscon";
 		#clock-cells = <1>;
 		clock-output-names = "ap-cpu-cluster-0", "ap-cpu-cluster-1",
-				     "ap-fixed", "ap-mss", "ap-emmc", "ap-hclk";
+				     "ap-fixed", "ap-mss", "ap-emmc", "dp-hclk";
 		reg = <0x6f4000 0x1000>;
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 00f92c3..5dd54f5 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -199,7 +199,7 @@
 				compatible = "marvell,coresight-axim", "arm,primecell";
 				reg = <0x840000 0x1000>;
 				clocks = <&syscon 3>, <&syscon 5>;
-				clock-names = "apb_pclk", "hclk";
+				clock-names = "apb_pclk", "dclk";
 				bus-width = <40>;
 			};
 
@@ -207,7 +207,7 @@
 				compatible = "marvell,coresight-axim", "arm,primecell";
 				reg = <0x841000 0x1000>;
 				clocks = <&syscon 3>, <&syscon 5>;
-				clock-names = "apb_pclk", "hclk";
+				clock-names = "apb_pclk", "dclk";
 				bus-width = <40>;
 			};
 
@@ -329,7 +329,7 @@
 				compatible = "marvell,ap806-system-controller", "syscon";
 				#clock-cells = <1>;
 				clock-output-names = "ap-cpu-cluster-0", "ap-cpu-cluster-1",
-						     "ap-fixed", "ap-mss", "ap-emmc", "ap-hclk";
+						     "ap-fixed", "ap-mss", "ap-emmc", "ap-dclk";
 				reg = <0x6f4000 0x1000>;
 			};
 
diff --git a/drivers/clk/mvebu/ap806-system-controller.c b/drivers/clk/mvebu/ap806-system-controller.c
index 8c90fb8..47c40e8 100644
--- a/drivers/clk/mvebu/ap806-system-controller.c
+++ b/drivers/clk/mvebu/ap806-system-controller.c
@@ -35,7 +35,7 @@ static struct clk_onecell_data ap806_clk_data = {
 
 static void __init ap806_syscon_clk_init(struct device_node *np)
 {
-	unsigned int freq_mode, cpuclk_freq, hclk_freq;
+	unsigned int freq_mode, cpuclk_freq, dclk_freq;
 	const char *name, *fixedclk_name;
 	struct regmap *regmap;
 	u32 reg;
@@ -85,46 +85,46 @@ static void __init ap806_syscon_clk_init(struct device_node *np)
 		pr_err("invalid SAR value\n");
 	}
 
-	/* Get HCLK frequency */
+	/* Get DCLK frequency */
 	switch (freq_mode) {
 	case 0x4:
 	case 0x10:
 	case 0x14:
 	case 0x19 ... 0x1D:
-		hclk_freq = 400;
+		dclk_freq = 400;
 		break;
 	case 0xC:
-		hclk_freq = 600;
+		dclk_freq = 600;
 		break;
 	case 0xD:
 	case 0x16:
-		hclk_freq = 525;
+		dclk_freq = 525;
 		break;
 	case 0xB:
 	case 0xE:
 	case 0xF:
-		hclk_freq = 450;
+		dclk_freq = 450;
 		break;
 	case 0x12:
 	case 0x13:
 	case 0x17:
-		hclk_freq = 325;
+		dclk_freq = 325;
 		break;
 	case 0x11:
 	case 0x15:
-		hclk_freq = 800;
+		dclk_freq = 800;
 		break;
 	case 0x18:
-		hclk_freq = 650;
+		dclk_freq = 650;
 		break;
 	default:
-		hclk_freq = 0;
+		dclk_freq = 0;
 		pr_err("invalid SAR value\n");
 	}
 
 	/* Convert to hertz */
 	cpuclk_freq *= 1000 * 1000;
-	hclk_freq *= 1000 * 1000;
+	dclk_freq *= 1000 * 1000;
 
 	/* CPU clocks depend on the Sample At Reset configuration */
 	of_property_read_string_index(np, "clock-output-names",
@@ -158,7 +158,7 @@ static void __init ap806_syscon_clk_init(struct device_node *np)
 	of_property_read_string_index(np, "clock-output-names",
 				      5, &name);
 	ap806_clks[5] = clk_register_fixed_rate(NULL, name, NULL, CLK_IS_ROOT,
-						hclk_freq);
+						dclk_freq);
 
 	of_clk_add_provider(np, of_clk_src_onecell_get, &ap806_clk_data);
 }
-- 
2.7.4

